backend/app/services/session_turn_service.py

```diff
@@ -37,6 +41,33 @@ def get_audio_content_type(upload_file: UploadFile) -> str:
     return mime
 
 
+def store_audio_file(session_id: UUID, audio_file: UploadFile) -> str:
+    audio_name = f'{session_id}_{uuid4().hex}'
+    gcs = GCSManager('audio')
+
+    try:
+        gcs.upload_from_fileobj(
+            file_obj=audio_file.file,
+            blob_name=audio_name,
+            content_type=get_audio_content_type(audio_file),
+        )
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f'Failed to upload audio file: {str(e)}') from e
```

Comment:
It would be best not to expose internal error messages to the user. It's a potential security risk.